packages/language-picker/src/language-picker.tsx

```diff
@@ -0,0 +1,109 @@
+/**
+ * External dependencies
+ */
+import React, { useState } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+export type LanguageGroup = {
+	id: string;
+	name: () => string;
```

Comment:
Unfortunately, we're going to hit into an issue with this now:

![](https://cldup.com/tzwBx5CJ3D.png)

Because the existing `LANGUAGE_GROUPS` constant expects that their name will get a translation function as an argument, and we're no longer passing it here, it's going to fail.

We should either:

* Address this in the constant definition, by using `__()` there directly, vs expecting a translation function. Can you think of any downside to this approach, @Automattic/i18n @yuliyan?
* Create a wrapper constant that uses `LANGUAGE_GROUPS` but predefines the `name` by passing `__()` as an argument to the function.

I personally prefer the first approach - ideally, it shouldn't be too hard to replace the instances of `languageGroup.name( translate )` to `languageGroup.name( `__` ) - I expect those to be only in the language picker component anyway.

Reply:
That makes sense (I thought I had fixed this but I guess I got the Storybook example and the live constants confused 😖)

Something I'm still confused about is whether we need to include the text domain global variable as described by @alshakero in the recent P2 here: pbAok1-1Ao-p2

If we do need to include the `__i18n_text_domain__` variable in the component (both here and in search) then do we need to include it in these constants as well?

Reply:
@tyxla After making this change actually there is a new but subtle bug. When selecting English from the langauge picker in Gutenboarding (accessed via `/new?flags=gutenboarding/language-picker`) it does not change the region languages to English (whereas for other languages it does work and in production it indeed will change to English when English is selected).

To reproduce:
1. Go to `/new?flags=gutenboarding/language-picker` on this branch
2. Click the language picker from the upper right hand corner
3. Select a languag other than English. Reopen the language picker and note that the regions are now listed in the langauge you selected.
4. Now select English. Reopen the language picker and note that the regions are still listed in the previous language you selected.
5. Repeat this several times going between languages that are not English and note that the regions change every time to the newly selected language _except for when you choose English_.

In production selecting English will behave the same as the rest of the languages.

Reply:
> Something I'm still confused about is whether we need to include the text domain global variable as described by alshakero in the recent P2 here: pbAok1-1Ao-p2

@Automattic/ganon are exploring these questions currently.

Reply:
> @Automattic/ganon are exploring these questions currently.

Latest WIP from @p-jackson here https://github.com/Automattic/wp-calypso/pull/46557